From 77636111ea2e0e50e1d4c3914311f7bd88620d97 Mon Sep 17 00:00:00 2001
From: Vincent Chen <vincent.chen@sifive.com>
Date: Fri, 26 Mar 2021 14:07:58 +0800
Subject: [PATCH 45/50] board: sifive: Set LED's color to blue before jumping
 to Linux

The LED's color wil be changed from purple to blue before executing
the sysboot command. Because the sysboot command includes the image loading
from the boot partition, It means the LED's color is blue when executing
"Retrieving file: /Image.gz".
---
 include/configs/sifive-hifive-unmatched-fu740.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/configs/sifive-hifive-unmatched-fu740.h b/include/configs/sifive-hifive-unmatched-fu740.h
index deb34142fc..7a1bdb877b 100644
--- a/include/configs/sifive-hifive-unmatched-fu740.h
+++ b/include/configs/sifive-hifive-unmatched-fu740.h
@@ -84,7 +84,12 @@
 	"type_guid_gpt_loader2=" TYPE_GUID_LOADER2 "\0" \
 	"type_guid_gpt_system=" TYPE_GUID_SYSTEM "\0" \
 	"partitions=" PARTS_DEFAULT "\0" \
-	BOOTENV
+	"setled_blue=mw.l 0x10020024 0x0000ffff; mw.l 0x10020028 0x0000ffff; mw.l 0x1002002c 0x0\0" \
+	BOOTENV \
+	"boot_extlinux="                                                  			\
+		"run setled_blue; "								\
+		"sysboot ${devtype} ${devnum}:${distro_bootpart} any "    			\
+			"${scriptaddr} ${prefix}${boot_syslinux_conf};\0"
 
 #define CONFIG_PREBOOT \
 	"setenv fdt_addr ${fdtcontroladdr};" \
-- 
2.31.0

